### PR TITLE
fix(client): Make json protocol work with data proxy

### PR DESCRIPTION
**Pull request by SevInf**
1. If JSON protocol is used, request was not serialized correctly and
   did not match either of 2 exepected bodies
2. Data proxy tests were not running on CI for JSON protocol, fixes that
   as well.

### CODE REVIEW

File: packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -369,7 +377,10 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
         }
 
         this._baseDmmf = new BaseDMMFHelper(config.document)
-        const engineProtocol = getQueryEngineProtocol(config.generator)
+        const engineProtocol = NODE_CLIENT
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:380]: Could we pass `config` to `getQueryEngineProtocol` and move this logic in there along with the other logic?

Comment (millsp) [reply packages/client/src/runtime/getPrismaClient.ts:380]: (In theory, NODE_CLIENT should be able to work there too, I think)

Comment (SevInf) [reply packages/client/src/runtime/getPrismaClient.ts:380]: `getQueryEngineProtocol` lives in `@prisma/internals` and is not build with `NODE_CLIENT` defined.

File: packages/client/src/runtime/getPrismaClient.ts [hunk 2]
```diff
@@ -244,6 +245,13 @@ export interface GetPrismaClientConfig {
    */
   injectableEdgeEnv?: LoadedEnv
 
+  /**
+   * Engine protocol to use within edge runtime. Passed
+   * through config because edge client can not read env variables
+   * @remarks only used for the purpose of data proxy
+   */
+  edgeClientProtocol?: QueryEngineProtocol
```
[the remark this answers is not in the document]
Comment (SevInf) [reply packages/client/src/runtime/getPrismaClient.ts:253]: It was not exactly silent, it was just that we were not running data proxy/edge client tests on JSON protocol. As soon we enabled that, this started failing pretty loudly.

File: packages/client/src/runtime/getPrismaClient.ts (hunk 1 above)
Comment (SevInf) [reply packages/client/src/runtime/getPrismaClient.ts:380]: `getQueryEngineProtocol` lives in `@prisma/internals`. Wouldn't using `NODE_CLIENT` there mean we'd have to build a separate edge version of internals?

Comment (millsp) [reply packages/client/src/runtime/getPrismaClient.ts:380]: Somehow I thought that esbuild would inject across the whole bundle, but that does not work.

**Review by millsp — APPROVED** [review]
✨

File: packages/client/src/runtime/getPrismaClient.ts (hunk 1 above)
Comment (millsp) [reply packages/client/src/runtime/getPrismaClient.ts:380]: Otherwise, I guess you could pass the `NODE_CLIENT` value as a param

**Pull request merged**